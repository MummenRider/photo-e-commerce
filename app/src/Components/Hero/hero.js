import React from "react";
import {
  HeroBackground,
  HeroContent,
  HeroH1,
  HeroLine,
  ImageBackground,
  Section,
} from "Components/Hero/hero.styles.js";
import heroImg from "Assets/Images/hero-bg@2x.png";

const HeroSection = () => {
  return (
    <Section data-testid="hero-container">
      <HeroBackground>
        <ImageBackground src={heroImg} alt="image-background" />
      </HeroBackground>
      <HeroContent>
        <HeroH1>Photography is the story I fail to put into words.</HeroH1>
      </HeroContent>
      <HeroLine data-testid="explore-line" />
    </Section>
  );
};

export default HeroSection;
