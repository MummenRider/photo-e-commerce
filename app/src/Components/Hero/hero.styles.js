import { motion } from "framer-motion";
import styled from "styled-components";

export const Container = styled(motion.section)`
  height: 100vh;
  min-height: 500px;
  display: flex;
  position: relative;
  justify-content: center;
  align-items: center;
`;

export const ImageFrame = styled.div`
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  overflow: hidden;
  height: 100%;
  width: 100%;
`;

export const Image = styled.img`
  height: 100%;
  width: 100%;
  object-fit: cover;
  filter: brightness(50%);
`;

export const Text = styled(motion.span)`
  font-family: lust-sans, sans-serif;
  font-size: clamp(30px, 5vw, 80px);
  font-style: italic;
  font-weight: 400;
  letter-spacing: 2px;
  color: white;
  padding: 10px 0;
`;
export const TextContainer = styled.h1`
  z-index: 2;
  display: flex;
  justify-content: center;
  flex-direction: column;
  width: 80%;
  margin: 0 auto;
  text-align: center;
`;
