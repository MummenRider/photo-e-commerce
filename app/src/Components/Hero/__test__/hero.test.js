import React from "react";
import { render, screen } from "@testing-library/react";
import Hero from "Components/Hero/hero.js";
import heroImg from "Assets/Images/hero-bg@2x.png";

it("should render hero component", () => {
  const { container, getByText } = render(
    <Hero>
      <Hero.ImageFrame>
        <Hero.Image src={heroImg} alt="image-background" />
      </Hero.ImageFrame>
      <Hero.TextContainer>
        <Hero.Text>Hero section</Hero.Text>
      </Hero.TextContainer>
    </Hero>
  );

  expect(screen.getByAltText("image-background")).toBeTruthy();
  expect(getByText("Hero section")).toBeTruthy();
  expect(container.firstChild).toMatchSnapshot();
});
