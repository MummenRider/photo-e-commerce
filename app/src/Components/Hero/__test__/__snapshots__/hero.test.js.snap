// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render hero component 1`] = `
<section
  class="sc-bdnxRM kDgKQe"
  data-testid="hero-container"
>
  <div
    class="sc-gtsrHT hUrVza"
  >
    <img
      alt="image-background"
      class="sc-dlnjwi iGfLot"
      src="hero-bg@2x.png"
    />
  </div>
  <div
    class="sc-hKFxyN jhIZwn"
  >
    <h1
      class="sc-eCApnc qgAqt"
    >
      Photography is the story I fail to put into words.
    </h1>
  </div>
  <div
    class="sc-jSFjdj eivmlX"
    data-testid="explore-line"
  />
</section>
`;
